### PR TITLE
FastCGI: Explicitly set Content-Length #626

### PR DESCRIPTION
**Pull request by humboldtux**
I've tested it against #567  and #614, and it will help me solve it too.

### CODE REVIEW

File: middleware/fastcgi/fastcgi.go [hunk 1]
```diff
@@ -105,13 +106,19 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error)
 				return http.StatusBadGateway, err
 			}
 
+			// Write the response body to a buffer
+			// To explicitly set Content-Length
+			// For FastCGI app that don't set it
+			var buf bytes.Buffer
+			io.Copy(&buf, resp.Body)
+			w.Header().Set("Content-Length", strconv.Itoa(buf.Len()))
```
Comment (mholt) [inline middleware/fastcgi/fastcgi.go]: We should only set this header if it is not already set in the response.

**Comment by mholt** [comment]
Awesome, thank you! I'll merge this then I'm gonna move a couple lines around, then see if I have time to add a test case.

**Comment by humboldtux** [comment]
Oups: this was not what i wanted to push: i didn't wrap in the if statement the buffer

**Comment by mholt** [comment]
I haven't done that yet (still working on test cases), so if you want to, you may, or else I will later. Just let me know.

**Comment by humboldtux** [comment]
I've pushed to `humboldtux:fastcgi-cl` but it's not showing here (maybe because you closed the Issue?)

A lot of code deduplication in my PR, i don't like it.

**Comment by mholt** [comment]
No problem, I took care of it in 737c7c437204922822d94bf41d6f374ccb13b935

**Pull request merged**